nano/node/distributed_work.hpp

```diff
@@ -32,6 +32,14 @@ class work_peer_request final
 	boost::asio::ip::tcp::socket socket;
 };
 
+enum class work_generation_status
+{
+	ongoing,
+	success,
+	cancelled,
+	failure
+};
```

Comment:
Minor: Maybe move this inside class `distributed_work` which is the only customer, as to not pollute the `nano` namespace.

Reply:
Good idea; thanks!